### ISSUE TITLE
Proxy shuts off automatically

### ISSUE THREAD

**Issue by Grochoci** [rvm/rvm#1462]
I have followed a tutorial to the letter and i have checked multiple times if everything is done properly and have checked multiple issues to see if someone's relates to mine. I have found nothing to help me. I start up my proxy and it says it is up and running and then when i press the siri button on my phone the terminal flashes a bunch of code then turns off.

``` ruby
parallels@parallels-Parallels-Virtual-Platform:~/SiriProxy/SiriProxy$ rvmsudo siriproxy server
Warning: can not check `/etc/sudoers` for `secure_path`, falling back to call via `/usr/bin/env`, this breaks rules from `/etc/sudoers`. export rvmsudo_secure_path=1 to avoid the warning.Starting SiriProxy on 0.0.0.0:443..
SiriProxy up and running.
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002c80b20 @signature=3, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002c80968>, @zip_stream=#<Zlib::Deflate:0x00000002c80918>, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins laoded: [#<SiriProxy::Plugin::Example:0x00000002cd8960 @manager=#<SiriProxy::PluginManager:0x00000002c80440 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002cd8550 @signature=5, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002cd83e8>, @zip_stream=#<Zlib::Deflate:0x00000002cd83c0>, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins laoded: [#<SiriProxy::Plugin::Example:0x00000002cd79e8 @manager=#<SiriProxy::PluginManager:0x00000002cd7fb0 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002cd74c0 @signature=7, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002cd7358>, @zip_stream=#<Zlib::Deflate:0x00000002cd7330>, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins laoded: [#<SiriProxy::Plugin::Example:0x000000028373c0 @manager=#<SiriProxy::PluginManager:0x00000002cd6f70 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002cd87a8 @signature=9, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002cd9ea0>, @zip_stream=#<Zlib::Deflate:0x00000002cd9f18>, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins laoded: [#<SiriProxy::Plugin::Example:0x00000002cd5080 @manager=#<SiriProxy::PluginManager:0x00000002cd3370 @plugins=[...]>>]
[Info - iPhone] Received Object: LoadAssistant
/home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:217:in `encode': "\x86" on UTF-8 (Encoding::InvalidByteSequenceError)
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:217:in `charset_convert'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:442:in `string_to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbCFTypes.rb:48:in `to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:550:in `block in dict_to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:550:in `map'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:550:in `dict_to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbCFTypes.rb:243:in `to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:550:in `block in dict_to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:550:in `map'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:550:in `dict_to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbCFTypes.rb:243:in `to_binary'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbBinaryCFPropertyList.rb:70:in `to_str'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbCFPropertyList.rb:363:in `to_str'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/CFPropertyList-2.2.0/lib/rbCFPropertyList.rb:398:in `to_plist'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/connection.rb:150:in `inject_object_to_output_stream'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/connection.rb:90:in `process_compressed_data'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/connection.rb:58:in `receive_binary_data'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/eventmachine-1.0.0/lib/em/protocols/linetext2.rb:94:in `receive_data'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/eventmachine-1.0.0/lib/eventmachine.rb:187:in `run_machine'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/eventmachine-1.0.0/lib/eventmachine.rb:187:in `run'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy.rb:16:in `initialize'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/command_line.rb:96:in `new'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/command_line.rb:96:in `start_server'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/command_line.rb:85:in `run_server'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/lib/siriproxy/command_line.rb:37:in `initialize'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/bin/siriproxy:6:in `new'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/gems/siriproxy-0.3.2/bin/siriproxy:6:in `<top (required)>'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/bin/siriproxy:23:in `load'
    from /home/parallels/.rvm/gems/ruby-1.9.3-p374@SiriProxy/bin/siriproxy:23:in `<main>'
parallels@parallels-Parallels-Virtual-Platform:~/SiriProxy/SiriProxy$ 
```

Is the warning at the beginning right before starting the server an issue i should be aware of? is there anything i am missing or doing wrong? Sometimes i get a different turn out and it says 

``` ruby
zlib(finalizer): the stream was freed prematurely.
```

I dont know if this is another issue or not but both interchange without any change in code from me.

Thanks in advance!
